dissect/target/loaders/mqtt.py

```diff
@@ -423,13 +426,147 @@ def connect(self) -> None:
         self.mqtt_client.loop_start()
 
 
-@arg("--mqtt-peers", type=int, dest="peers", help="minimum number of peers to await for first alias")
-@arg("--mqtt-case", dest="case", help="case name (broker will determine if you are allowed to access this data)")
-@arg("--mqtt-port", type=int, dest="port", help="broker connection port")
-@arg("--mqtt-broker", dest="broker", help="broker ip-address")
-@arg("--mqtt-key", dest="key", help="private key file")
-@arg("--mqtt-crt", dest="crt", help="client certificate file")
-@arg("--mqtt-ca", dest="ca", help="certificate authority file")
+def strictly_positive(value: str) -> int:
+    """
+    Validates that the provided value is a strictly positive integer.
+
+    This function is intended to be used as a type for argparse arguments.
+
+    Args:
+        value (str): The value to validate.
+
+    Returns:
+        int: The validated integer value.
+
+    Raises:
+        argparse.ArgumentTypeError: If the value is not a strictly positive integer.
+    """
+    try:
+        peer_value = int(value)
+        if peer_value < 1:
+            raise argparse.ArgumentTypeError("Value must be strictly positive.")
+        return peer_value
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"Invalid integer value specified: '{value}'")
+
+
+def host_name(value: str) -> bool:
+    """Checks if the value is a valid host name."""
+
+    # Adapted from https://stackoverflow.com/questions/2532053/validate-a-hostname-string
+    if value[-1] == ".":
+        # Host names are allowed to end in a dot
+        value = value[:-1]
+    if len(value) > 253:
+        return False
+
+    labels = value.split(".")
+
+    # The top level domain must contain at least one alphabetic character.
+    # Stricter than necessary since a TLD can consists of multiple labels.
+    if re.match(r"[0-9]+$", labels[-1]):
+        return False
+
+    # Each label must be between 1 and 63 characters long.
+    # Labels cannot begin or start with a hyphen.
+    label_pattern = re.compile(r"(?!-)[a-z0-9-]{1,63}(?<!-)$", re.IGNORECASE)
+    return all(label_pattern.match(label) for label in labels)
+
+
+def ip_address(value: str) -> bool:
+    """Checks if the value is a valid IP address."""
+    try:
+        ipaddress.ip_address(value)
+        return True
+    except ValueError:
+        return False
+
+
+def host_name_or_ip_address(value: str) -> str:
```

Comment:
For reviewer: alternatively you could write a parser combinator 

`def or_parser(parser1: Callable[[str], T1], parser2: Callable[[str], T2]) -> Callable[[str], T1 | T2]` 

but this is easier to understand, and there are probably python libraries which do that much better.